src/KpmRepoManager.ts

```diff
@@ -1,185 +1,179 @@
+  if (projectDirs && projectDirs.length > 0) {
+    if (!fileName) {
+      return projectDirs[0];
     }
 
-    // VSCode allows having multiple workspaces.
-    // for now we only support using the 1st project directory
-    // in a given set of workspaces if the provided fileName is null.
-    if (projectDirs && projectDirs.length > 0) {
-        if (!fileName) {
-            return projectDirs[0];
-        }
-
-        for (let i = 0; i < projectDirs.length; i++) {
-            const dir = projectDirs[i];
-            if (fileName.includes(dir)) {
-                return dir;
-            }
-        }
+    for (let i = 0; i < projectDirs.length; i++) {
+      const dir = projectDirs[i];
+      if (fileName.includes(dir)) {
+        return dir;
+      }
     }
-    return null;
+  }
+  return null;
 }
 
 async function getCommandResult(cmd, projectDir) {
-    let result = await wrapExecPromise(cmd, projectDir);
-    if (!result) {
-        // something went wrong, but don't try to parse a null or undefined str
-        return null;
-    }
-    result = result.trim();
-    let resultList = result
-        .replace(/\r\n/g, "\r")
-        .replace(/\n/g, "\r")
-        .replace(/^\s+/g, " ")
-        .replace(/</g, "")
-        .replace(/>/g, "")
-        .split(/\r/);
-    return resultList;
+  let result = await wrapExecPromise(cmd, projectDir);
+  if (!result) {
+    // something went wrong, but don't try to parse a null or undefined str
+    return null;
+  }
+  result = result.trim();
+  let resultList = result
+    .replace(/\r\n/g, "\r")
+    .replace(/\n/g, "\r")
+    .replace(/^\s+/g, " ")
+    .replace(/</g, "")
+    .replace(/>/g, "")
+    .split(/\r/);
+  return resultList;
 }
 
 export async function getFileContributorCount(fileName) {
-    let fileType = getFileType(fileName);
+  let fileType = getFileType(fileName);
 
-    if (fileType === "git") {
-        return 0;
-    }
+  if (fileType === "git") {
+    return 0;
+  }
 
-    const projectDir = getProjectDir(fileName);
-    if (!projectDir) {
-        return 0;
-    }
+  const projectDir = getProjectDir(fileName);
+  if (!projectDir) {
+    return 0;
+  }
 
-    // all we need is the filename of the path
-    // const baseName = path.basename(fileName);
+  // all we need is the filename of the path
+  // const baseName = path.basename(fileName);
 
-    const cmd = `git log --pretty="%an" ${fileName}`;
+  const cmd = `git log --pretty="%an" ${fileName}`;
 
-    // get the list of users that modified this file
-    let resultList = await getCommandResult(cmd, projectDir);
-    if (!resultList) {
-        // something went wrong, but don't try to parse a null or undefined str
-        return 0;
+  // get the list of users that modified this file
+  let resultList = await getCommandResult(cmd, projectDir);
+  if (!resultList) {
+    // something went wrong, but don't try to parse a null or undefined str
+    return 0;
+  }
+
+  if (resultList.length > 0) {
```

Comment:
how about: 

```
function countUnique(iterable) {
  return new Set(iterable).size;
}

countUnique(resultList)
```

Reply:
i love this way of doing it. i'll replace that, thanks.

Reply:
added that function, thanks!